### ISSUE TITLE
evaluation hot keys should be disabled if a modal is open

### ISSUE THREAD

**Issue by bcolloran** [iodide-project/iodide#2100]
ctrl+enter and shift+enter currently work when a modal is open. we should intercept and block the evaluation action if a modal is open (i.e. if `modalState !== "MODALS_CLOSED"`)

**Comment by Algentile** [comment]
Hey @bcolloran,

I was able to reproduce this issue when the I select the editor while the modal is open. I can work on this.

**Comment by bcolloran** [comment]
great, thanks @Algentile. fwiw: i think the right place to put this is in the `evaluateText` redux thunk.

**Comment by Algentile** [comment]
Hey @bcolloran , 

I opened a pull request for the issue [#2154 ](https://github.com/iodide-project/iodide/pull/2154) for the change. I did notice a weird behavior in my testing.

It appears that if an NoteBookMenuitem has been previously selected and you use the shortcut `ctrl+enter`, it will open the menu item and run the evaluate shortcut. This does not happen if you use `shift+enter`, It seems that this might have been unintended and I was curious if I should create a separate issue for this behavior?
